pkg/cri/server/image_pull.go

```diff
@@ -113,12 +114,14 @@ func (c *criService) PullImage(ctx context.Context, r *runtime.PullImageRequest)
 		}
 	)
 
+	labels := c.getLabels(ref)
+
 	pullOpts := []containerd.RemoteOpt{
 		containerd.WithSchema1Conversion,
 		containerd.WithResolver(resolver),
 		containerd.WithPullSnapshotter(c.config.ContainerdConfig.Snapshotter),
 		containerd.WithPullUnpack,
-		containerd.WithPullLabel(imageLabelKey, imageLabelValue),
+		containerd.WithPullLabels(labels),
```

Comment:
> I'd like to discuss making this more generic... would like to consider using an image annotation .. For example, when pulling the pause image add an annotation that requests the image be flagged as pinned.

@mikebrow already doing what you suggested, adding an annotation when pause image is pulled, anything i am missing here?
```
func (c *criService) getLabels(name string) map[string]string {
	labels := map[string]string{imageLabelKey: imageLabelValue}
	// Adding sandbox image label to containerd store to avoid
	// new CRI field "Pinned" in containerd image store
	if c.config.SandboxImage == name {
		sandboxImageLabelKey, sandboxImageLabelValue := util.GetSandboxImageLabels()
		labels[sandboxImageLabelKey] = sandboxImageLabelValue
	}
	return labels
}
```

Reply:
```golang
if c.config.SandboxImage == name {
		sandboxImageLabelKey, sandboxImageLabelValue := util.GetSandboxImageLabels()
		labels[sandboxImageLabelKey] = sandboxImageLabelValue
	}
````

current implementation the label name and value is strongly related to  sandbox image,  how about make it more generic, just a label to indicate the image is pinned.

Reply:
missed your response when I was on vac..  

right.. so what I'm talking about is in PullImage() (inbound CRI pullrequest) check PullImageRequest.Image.Annotations[] for an annotation (maybe "k8s.io/cri/pinned"="true" or similar) indicating that this image when successfully pulled should be pinned

Reply:
in this fashion.. the caller to PullImage can identify if the image should carry the pinned attribute

on the other side in our store this annotation will already be there.. in the store image.. though you could also add a label for pinned as @scuzhanglei mentioned..

Reply:
A quick check in kubelet image_manager... the image annotations on the pull image request are the set of pod annotations.. I don't see another meta object injection method in kubelet that apples to images ..

Reply:
+1 on making this logic more generic than just sandbox image. I think there are two cases where containerd wants to pin an image:

a) the sandbox image. Containerd will just pin the image if the image name matches the sandbox image name.

b) some arbitrary image that the user wants to pin. The user (caller of pullImage) can add a `PullImageRequest.Image.Annotation` (maybe "k8s.io/cri/pinned"="true" or similar). Containerd will check this Annotation and mark the image as pinned. 

Seems like this PR supports case a only.